src/client/systems/ClientGameController.cpp

```diff
@@ -11,7 +11,15 @@
 #include "ClientGameController.hpp"
 
 ClientGameController::ClientGameController() :
-    GameController(),
+    m_headless(false), m_isBot(false), m_strategy(BotStrategy::Random), GameController(),
+    m_window(sf::VideoMode(WINDOW_RESOLUTION.x, WINDOW_RESOLUTION.y), "Ungroup", sf::Style::Close),
+    m_networkingClient(new NetworkingClient()), m_inputController(new InputController(INPUT_KEYS)),
+    m_renderingController(
+        new RenderingController(m_window, *m_gameObjectController, *m_resourceStore)) {
+}
+
+ClientGameController::ClientGameController(bool is_headless, bool is_bot, BotStrategy strategy) :
+    m_headless(is_headless), m_isBot(is_bot), m_strategy(strategy), GameController(),
     m_window(sf::VideoMode(WINDOW_RESOLUTION.x, WINDOW_RESOLUTION.y), "Ungroup", sf::Style::Close),
     m_networkingClient(new NetworkingClient()), m_inputController(new InputController(INPUT_KEYS)),
     m_renderingController(
```

Comment:
I don't like the amount of code duplicated here just to support different parameters 

You can just provide default parameters and have one constructor:
```cpp
bool is_headless = false, bool is_bot = false, BotStrategy strategy = BotStrategy::Random
```

Reply:
I didn't know C++ had default params, good call